lib/sqlalchemy/connectors/pyodbc.py

```diff
@@ -77,6 +77,9 @@ def create_connect_args(self, url: URL) -> ConnectArgsType:
         else:
 
             def check_quote(token: str) -> str:
+                # token is already quoted, so return it as-is
```

Comment:
But what if alice's password really is `{password}`? Should we expect her to ODBC-escape the password herself? Or, put another way, if she does "the right thing" (IMO) and uses `URL.create()` then the result is not correct:

```python
import sqlalchemy as sa
from sqlalchemy.dialects.mssql import pyodbc

connection_url = sa.engine.URL.create(
    "mssql+pyodbc",
    username="alice",
    password="{password}",
    host="localhost",
    database="mydb",
    query=dict(driver="ODBC Driver 17 for SQL Server")
)
dialect = pyodbc.dialect()
connection = dialect.create_connect_args(connection_url)
print(connection[0][0])
# DRIVER={ODBC Driver 17 for SQL Server};Server=localhost;Database=mydb;UID=alice;PWD={password}
```

It should be `PWD={{password}}}`

Reply:
Ok, yep - it's the balancing of these curly braces that was puzzling me here.  I do now see that `ESCAPEDRCB` exists in the language spec (where `RCB` = right curly bracket), and that there's no corresponding LCB.

I'm taking a bit more time to internalize and grok that, but initially I suppose it's reasonable.

Reply:
Ok, I think I've convinced myself that 8ac7cb92b4972a08b8008b80b34989694510139f is correct as-is; my apologies for the distractions here.

Something about the directional nature of the escaping (see a `{`, apply the `}}` rule on the suffix) seemed to make it tricky to mentally enumerate the possibilities and validate strings visually - but I do think the implementation is correct.  I'll likely close this pull request soon.

Thanks for the review and explanations.